### PR TITLE
feat: add PHP generator

### LINKED ISSUE #1195: PHP Generator Scoped
## Milestones
- [x] Base PHP Generator from the official template (https://github.com/asyncapi/modelina/pull/1196)
- [x] Define PHP reserved words/keywords (https://github.com/asyncapi/modelina/pull/1211)
- [x] Ability to generate classes
- [x] Ability to generate Enums
- [x] Mixed & Nullable Types (https://github.com/asyncapi/modelina/pull/1216)
- [x] Documentation
- [x] Examples (https://github.com/asyncapi/modelina/pull/1218)
- [x] PHP Namespace as a parameter (https://github.com/asyncapi/modelina/pull/1258)
- [x] Render strict types (https://github.com/asyncapi/modelina/pull/1253)
- [ ] Final classes by default and parameter

### Additionals
* How about Backed Enums? https://www.php.net/manual/en/language.enumerations.backed.php

## Thoughts, Ideas, PHP Specifics 

> Let's support only PHP 8.0 and up. PHP 7.4 is already not supported: https://www.php.net/supported-versions.php

### PHP 8.0
* private properties when readonly -> getters
* public properties instead of getters + setters

### PHP 8.1
* public readonly properties
* public properties instead of getters + setters

### PHP 8.2
* public readonly class
* public properties

## Options
* PHP Version (8.0, 8.1, 8.2) default (8.2)
* Readonly properties (true/false) default (true)
* Namespace

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR is a placeholder for when to release the PHP generator.

Resolves #1195 
Resolves https://github.com/asyncapi/modelina/issues/744

### CODE REVIEW

**Comment by jonaslagoni** [comment]
Hey @Ferror, just pinging you to ask what's missing here before we can merge it? 🤔 Does not seem like much to me 🤔

**Comment by Ferror** [comment]
> Hey @Ferror, just pinging you to ask what's missing here before we can merge it? 🤔 Does not seem like much to me 🤔

I'm missing a few details to release it, but I will be working on them next week. We can schedule some Grant Review 28th of May :)

**Comment by jonaslagoni** [comment]
> I'm missing a few details to release it, but I will be working on them next week. We can schedule some Grant Review 28th of May :)

Sounds great 💪 

Once the remaining PRs are merged we don't need to review this PR as it has no changes that have not been approved previously 🙂 So it's only a matter of seeing if we are missing anything, and then we can merge it 👍

**Comment by Ferror** [comment]
> > I'm missing a few details to release it, but I will be working on them next week. We can schedule some Grant Review 28th of May :)
> 
> Sounds great 💪
> 
> Once the remaining PRs are merged we don't need to review this PR as it has no changes that have not been approved previously 🙂 So it's only a matter of seeing if we are missing anything, and then we can merge it 👍

I mean April, not May. I don't think we will need to wait a month 😄

**Comment by Ferror** [comment]
When the [PR](https://github.com/asyncapi/modelina/pull/1258) we could ask our PHP friends to test the solution. I'm out of ideas what's should be next 😄

File: docs/languages/Php.md
Comment (jonaslagoni) [inline docs/languages/Php.md:4]: This is actually a lie, cause you have the description preset @Ferror 😄

**Review by jonaslagoni — COMMENTED** [review]
Ferror this is the only thing I think are missing:
- Description preset example.
- Blackbox tests for PHP generator (you can use this to inspect the output of real-world examples and see how they are generated in PHP)
- Description preset documentation for the PHP language.

**Comment by Ferror** [comment]
Hmm regarding the package-lock.json revert it's really not that easy as the master was already merged a couple of times into this PR. Can we rebase this branch on top of the master? And in case of any conflicts accept those from master branch

**Comment by jonaslagoni** [comment]
@Ferror I solved all the merge conflicts in https://github.com/asyncapi/modelina/pull/1205/commits/192d0c809863c0a204b7edbbad9c5b6a0a49f935 🙂

**Comment by jonaslagoni** [comment]
The only thing left is this:
- Description preset example.
- Description preset documentation for the PHP language.

And then it's merge time 😀

**Comment by jonaslagoni** [comment]
@Ferror had to make a few adjustments, see the recent commits, LGTM 👍 

@magicmatatjahu @kennethaasan can I get a quick 👍 here 🙂

**Review by kennethaasan — APPROVED** [review]
Had a quick look. LGTM

**Comment by jonaslagoni** [comment]
Thanks @kennethaasan!

@Ferror as always, as the creator of the generator, if you would like to long term maintain the PHP generator, feel free to add a PR adding yourself as [codeowner](https://github.com/asyncapi/modelina/blob/master/CODEOWNERS), and I can invite you on as maintainer. That is if you want to 🙂

Merging.

**Pull request merged**
